src/DbExportsRegistry.cpp

```diff
@@ -0,0 +1,75 @@
+#include "DbExportsRegistry.h"
+
+#include <QDate>
+#include <QSettings>
+
+DbExportsRegistry::DbExportsRegistry(const QString &settingsPath, QObject *parent) :
+    QObject(parent), settingsPath(settingsPath)
+{
+}
+
+void DbExportsRegistry::setCurrentCardDbMetadata(QString cardId, int credentialsDbChangeNumber, int dataDbChangeNumber)
+{
+    DbExportsRegistry::cardId = cardId;
+    DbExportsRegistry::credentialsDbChangeNumber = credentialsDbChangeNumber;
+    DbExportsRegistry::dataDbChangeNumber = dataDbChangeNumber;
+
+    checkIfDbMustBeExported();
+}
+
+void DbExportsRegistry::checkIfDbMustBeExported()
+{
+    if (!cardId.isEmpty())
+    {
+        const QDate date = getLastExportDate(cardId);
+        const int lastCredentialCN = getLastExportCredentialDbChangeNumber(cardId);
+        const int lastDataCN = getLastExportDataDbChangeNumber(cardId);
+
+        if  (isOlderThanAMonth(date) &&
+             (lastCredentialCN < credentialsDbChangeNumber || lastDataCN < dataDbChangeNumber))
```

Comment:
anddddddd you're not handling the wrapover. please use the code you actually made.